analyzer/src/main/resources/UserAgents/TV.yaml

```diff
@@ -2449,3 +2459,162 @@ config:
       AgentNameVersion                     : 'Unknown ??'
       AgentNameVersionMajor                : 'Unknown ??'
       AgentSecurity                        : 'Strong security'
+
+### VIZIO TV Detection
+
+# VIZIO smart cast and older TV detection
+- matcher:
+    variable:
+      - 'VTAG                              :agent.product.name{"VIZIO"^'
+    extract:
+      - 'DeviceClass                         :     2000 :"TV"'
+      - 'DeviceName                          :     2000 :@VTAG.name'
+      - 'DeviceBrand                         :     2000 :"Vizio"'
+      - 'DeviceVersion                       :     2000 :@VTAG.(1)comments.(1)entry.(3)product.(1)version'
+      - 'OperatingSystemVersion              :     2000 :@VTAG.(1)comments.(1)entry.(2)product.(1)version'
+      - 'OperatingSystemClass                :     2000 :"Embedded"'
+      - 'OperatingSystemName                 :     2000 :"Vizio Smartcast"'
+      - 'OperatingSystemVersion              :     2000 :"<<<null>>>"'
+      - 'OperatingSystemVersionMajor         :     2000 :"<<<null>>>"'
+
+- matcher:
+    variable:
+      - 'VTAG                              :agent.product.name{"VIZIO"^'
+    require:
+      - '@VTAG.(1)comments.(1)entry.(1)text="Vizio"'
+    extract:
+      - 'DeviceClass                         :     2001 :"TV"'
+      - 'DeviceName                          :     2001 :@VTAG.name'
+      - 'DeviceBrand                         :     2001 :"Vizio"'
+      - 'DeviceVersion                       :     2001 :@VTAG.(1)comments.(2)entry'
+      - 'OperatingSystemVersion              :     2001 :@VTAG.version'
+      - 'OperatingSystemClass                :     2001 :"Embedded"'
+      - 'OperatingSystemName                 :     2001 :"Vizio DTV"'
+      - 'OperatingSystemVersion              :     2001 :"<<<null>>>"'
+      - 'OperatingSystemVersionMajor         :     2001 :"<<<null>>>"'
+
+- test:
+    input:
+      # The parser complains of an Ambiguity in the UA which causes the test to fail. I'm not sure how to address since I can't
+      # control the incoming UA. The parsing appears to work correctly
```

Comment:
what should be done with the ambiguity complaints, it wasn't obvious to me if it was the rules or the actual UA string that it was complaining about.

Reply:
Useragents are (by definition almost) a mess. The "ambiguity" messages come from the way the underlying parser has been constructed. You can ignore those.